app/controllers/visual_groups/images_controller.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module VisualGroups
+  class ImagesController < ApplicationController
+    before_action :login_required
+
+    def update
+      @user = User.current
+      image_id = params[:id]
+      @status = params[:status]
+      @status = "" unless Image.find_by(id: image_id)
+      visual_group = VisualGroup.find_by(id: params[:visual_group_id])
+      return unless visual_group
+
+      status = update_visual_group_image(visual_group, image_id)
+
+      # this is a turbo response
+      render(partial: "visual_groups/images/update",
+             locals: { status: status, image_id: image_id,
+                       visual_group: visual_group })
+    end
+
+    def update_visual_group_image(visual_group, image_id)
+      vgi = visual_group.visual_group_images.find_by(image_id: image_id)
+      included = (@status == "true")
+      if @status == ""
+        vgi&.destroy
```

Comment:
The problem is when you click 'Needs Review'.  It should just get selected (become dark and bold), but on your branch it selects 'Exclude' instead.  I haven't checked what it does on the backend.  It is intentional that the page doesn't otherwise change so you can power through lots of images without stuff jumping around.  It also makes it more forgiving if you make a mistake.

Reply:
Got it, tx for the quick response.

Reply:
Ah, the issue is that private method `update_visual_group_image` is coercing the value of `status` to a boolean, when it needs to preserve the `nil` value for `needs_review`. 

The method doesn't seem any different from `main`... will debug further.

Reply:
Found my bug. Should an image marked "Needs Review" still appear in the visual group on reload, and only the "Excluded" images fall out of the group? 

On `main` the "Needs Review" images do not appear in the group on reload, but then i'm wondering how you would eventually "review" those images if they don't appear in the group.

(I missed the coercion to string of `params[:value].to_s` so i was checking for `""` instead of `nil`.)

Reply:
Thank you, that helps!

However, on `main`, just now, I do *not* find the "Needs Review" images on the Edit view.
I *do* find the "Excluded" images when I click that link, and also the "Included". But even when returning to "Needs Review" via the link that appears on "Excluded", no images appear. 

I know you've been using this UI, and it must work, but that doesn't seem right. I notice that in `AjaxController::VisualGroupStatus`, `vgi&.destroy` is only called on a VisualGroupImage with status == "", i.e. "Needs Review". Can you confirm this is correct?

Reply:
Ok as far as I can tell there are two things that keep VisualGroupImages from behaving as described. One is that the status values map like this:
```
needs_review | included | excluded
nil                         true           false
```
but having a `needs_review` value is the one that gets the VGI deleted, where I would expect it to be the `excluded` value.
```ruby
    if @value == ""
      vgi&.destroy
      status = nil
```

The second issue is that the VisualGroupImages table does not allow nil values on the `included` column.
```
    t.boolean "included", default: true, null: false
```

Reply:
@mo-nathan - Just want to be sure you saw this question about the setup of `VisualGroupImages`. No rush, but when you have time to consider, please let me know what you think. 

I think the controller behavior should maybe be changed.

Reply:
Resolved via email - this is a bug on `main` and will leave as-is on this PR.